ui/admin/tests/acceptance/auth-methods/oidc-test.js

```diff
@@ -0,0 +1,139 @@
+  setupMirage(hooks);
+
+  const instances = {
+    scopes: {
+      global: null,
+      org: null,
+    },
+    authMethod: null,
+  };
+  const urls = {
+    orgScope: null,
+    authMethods: null,
+    authMethod: null,
+  };
+
+  hooks.beforeEach(function () {
+    authenticateSession({});
+    instances.scopes.global = this.server.create('scope', { id: 'global' });
+    instances.scopes.org = this.server.create('scope', {
+      type: 'org',
+      scope: { id: 'global', type: 'global' },
+    });
+    instances.authMethod = this.server.create('auth-method', {
+      scope: instances.scopes.org,
+      type: 'oidc',
+    });
+
+    urls.orgScope = `/scopes/${instances.scopes.org.id}`;
+    urls.authMethods = `${urls.orgScope}/auth-methods`;
+    urls.authMethod = `${urls.authMethods}/${instances.authMethod.id}`;
+  });
+
+  test('visiting oidc auth method', async function (assert) {
+    assert.expect(1);
+    await visit(urls.authMethod);
+    await a11yAudit();
+    assert.equal(currentURL(), urls.authMethod);
+  });
+
+  test('can view oidc state', async function (assert) {
+    assert.expect(1);
+    await visit(urls.authMethod);
+    await click('.rose-layout-page-actions .rose-dropdown-trigger');
+    assert.equal(
+      find('.rose-dropdown[open] input[aria-checked=true]').value,
+      instances.authMethod.attributes.state
+    );
+  });
+
+  test('can update oidc state', async function (assert) {
+    assert.expect(1);
+    const updateValue = 'inactive';
+    await visit(urls.authMethod);
+    await click('.rose-layout-page-actions .rose-dropdown-trigger');
+    await click(`.rose-dropdown[open] input[value="${updateValue}"]`);
+    const authMethod = this.server.db.authMethods.find(instances.authMethod.id);
+    assert.equal(authMethod.attributes.state, updateValue);
+  });
+
+  test('can update oidc state to active-private', async function (assert) {
+    assert.expect(1);
+    const updateValue = 'active-private';
+    await visit(urls.authMethod);
+    await click('.rose-layout-page-actions .rose-dropdown-trigger');
+    await click(`.rose-dropdown[open] input[value="${updateValue}"]`);
+    const authMethod = this.server.db.authMethods.find(instances.authMethod.id);
+    assert.equal(authMethod.attributes.state, updateValue);
+  });
+
+  test('can update oidc state to active-public', async function (assert) {
+    assert.expect(1);
+    // Update default 'active-public' state to inactive
+    instances.authMethod.attributes.state = 'inactive';
+    const updateValue = 'active-public';
+    await visit(urls.authMethod);
+    await click('.rose-layout-page-actions .rose-dropdown-trigger');
+    await click(`.rose-dropdown[open] input[value="${updateValue}"]`);
+    const authMethod = this.server.db.authMethods.find(instances.authMethod.id);
+    assert.equal(authMethod.attributes.state, updateValue);
+  });
+
+  // FIXME: How to mock just one request routed to /auth-methods and let everything else passthrough?
```

Comment:
Shall we leave this commented to be fixed for next release?

Reply:
I'll circle back to this in another PR.